pkg/api/metrics.go

```diff
@@ -28,6 +29,12 @@ func (hs *HTTPServer) handleQueryMetricsError(err error) *response.NormalRespons
 	if errors.Is(err, plugincontext.ErrPluginNotFound) {
 		return response.Error(http.StatusNotFound, "Plugin not found", err)
 	}
+	if errors.Is(err, plugins.ErrPluginNotRegistered) {
```

Comment:
Ugh I feel like there's so many different errors for when a plugin doesn't exist. Do you think there any way we can have just one `plugins.ErrPluginNotFound` that can be used everywhere?

Reply:
I agree, that's why I added a comment for removing at least `plugincontext.ErrPluginNotFound` but didn't want to add more things to this PR. I can do that in a follow-up PR.